bioio/writers/ome_zarr_writer.py

```diff
@@ -1,431 +1,478 @@
+                )
+                ch = {
+                    "active": True,
+                    "coefficient": 1.0,
+                    "color": color,
+                    "family": "linear",
+                    "window": {
+                        "min": float(win_min),
+                        "max": float(win_max),
+                        "start": float(win_min),
+                        "end": float(win_max),
+                    },
+                    "label": label,
+                }
+                channel_meta.append(ch)
 
-        # Save image to zarr store!
-        self.store = parse_url(uri, mode="w").store
-        self.root_group = zarr.group(store=self.store)
+        # Write metadata
+        self._write_metadata(image_name, axes_meta, channel_meta, creator_info)
 
-    @staticmethod
-    def build_ome(
-        size_z: int,
-        image_name: str,
-        channel_names: List[str],
-        channel_colors: List[int],
-        channel_minmax: List[Tuple[float, float]],
-    ) -> Dict:
+    def _compute_levels(self, max_levels: Optional[int]) -> List[Tuple[int, ...]]:
         """
-        Create the omero metadata for an OME zarr image
+        Calculate all multiresolution level shapes by repeatedly scaling.
+        Minimum dimension size will always be 1 and stops when no further reduction
+        or at max_levels.
 
         Parameters
         ----------
-        size_z:
-            Number of z planes
-        image_name:
-            The name of the image
-        channel_names:
-            The names for each channel
-        channel_colors:
-            List of all channel colors
-        channel_minmax:
-            List of all (min, max) pairs of channel intensities
+        max_levels : Optional[int]
+            Number of levels to compute (including base level). None for full descent.
 
         Returns
         -------
-        Dict
-            An "omero" metadata object suitable for writing to ome-zarr
+        List[Tuple[int, ...]]
+            List of 5D shape tuples for each pyramid level.
         """
-        ch = []
-        for i in range(len(channel_names)):
-            ch.append(
-                {
-                    "active": True,
-                    "coefficient": 1,
-                    "color": f"{channel_colors[i]:06x}",
-                    "family": "linear",
-                    "inverted": False,
-                    "label": channel_names[i],
-                    "window": {
-                        "end": float(channel_minmax[i][1]),
-                        "max": float(channel_minmax[i][1]),
-                        "min": float(channel_minmax[i][0]),
-                        "start": float(channel_minmax[i][0]),
-                    },
-                }
+        shapes = [self.shape]
+        lvl = 1
+        while max_levels is None or lvl < max_levels:
+            prev = shapes[-1]
+            nxt = tuple(
+                max(1, prev[i] // self.scale_factors[i])
```

Comment:
We previously let users choose scale factors for each level (like 0.5, 0.4, 0.3, 0.2) idk how useful that is but is there a reason you removed that?